src/appleseed/foundation/meta/tests/test_iesparser.cpp

```diff
@@ -0,0 +1,313 @@
+        parser.parse_keyword_line(parser.line);
+        parser.read_trimmed_line(input_stream);
+        EXPECT_EXCEPTION(IESParser::ParsingException, parser.check_required_keywords());
+        parser.parse_keyword_line(parser.line);
+        parser.read_trimmed_line(input_stream);
+        parser.check_required_keywords();
+
+        parser.format = IESParser::Format2002;
+        parser.parse_keyword_line(parser.line);
+        parser.read_trimmed_line(input_stream);
+        EXPECT_EXCEPTION(IESParser::ParsingException, parser.check_required_keywords());
+        parser.parse_keyword_line(parser.line);
+        parser.read_trimmed_line(input_stream);
+        parser.check_required_keywords();
+
+        parser.parse_keyword_line(parser.line);
+        parser.read_trimmed_line(input_stream);
+        parser.parse_keyword_line(parser.line);
+        parser.read_trimmed_line(input_stream);
+        parser.parse_keyword_line(parser.line);
+        parser.read_trimmed_line(input_stream);
+
+        EXPECT_EXCEPTION(IESParser::ParsingException, parser.parse_keyword_line(parser.line));
+        parser.read_trimmed_line(input_stream); // Bad keyword
+        EXPECT_EXCEPTION(IESParser::ParsingException, parser.parse_keyword_line(parser.line));
+        parser.read_trimmed_line(input_stream); // Not a keyword-value pair
+        EXPECT_EXCEPTION(IESParser::ParsingException, parser.parse_keyword_line(parser.line));
+        parser.read_trimmed_line(input_stream); // EOF
+        EXPECT_EXCEPTION(IESParser::ParsingException, parser.parse_keyword_line(parser.line));
+
+        EXPECT_EQ(parser.keywords_dictionary.at("KEYWORD"), "Value");
+        EXPECT_EQ(parser.keywords_dictionary.at("MANUFAC"), "Manufac is required by 1991 and 2002");
+        EXPECT_EQ(parser.keywords_dictionary.at("_USER"), "User\nMore\nMore");
+        EXPECT_EXCEPTION(std::out_of_range, parser.keywords_dictionary.at("BAD"));
+        EXPECT_EXCEPTION(std::out_of_range, parser.keywords_dictionary.at("MORE"));
+    }
+
+    TEST_CASE(ParseToVector_Empty)
+    {
+        IESParser parser;
+        {
+            std::istringstream input_stream("");
+            parser.reset(input_stream);
+            std::vector<int> result = parser.parse_to_vector<int>(input_stream, 0);
+            EXPECT_EQ(result, std::vector<int>());
+        }
+        {
+            std::istringstream input_stream("test");
+            parser.reset(input_stream);
+            std::vector<double> result = parser.parse_to_vector<double>(input_stream, 0);
+            EXPECT_EQ(parser.line, "test");
+            EXPECT_EQ(result, std::vector<double>());
+        }
+    }
+
+    TEST_CASE(ParseToVector_Good)
+    {
+        IESParser parser;
+        {
+            std::istringstream input_stream("\n-0 1    2  3 \n\t4");
+            parser.reset(input_stream);
+            std::vector<int> result = parser.parse_to_vector<int>(input_stream, 5);
+            EXPECT_EQ(result.size(), 5);
```

Comment:
This is a typical case where using `ASSERT_EQ()` is recommended: it will stop the test if `result.size()` is not equal to 5 (because in that case you cannot safely continue).